lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -1656,6 +1834,9 @@ SILInstruction *SILCombiner::visitApplyInst(ApplyInst *AI) {
           return nullptr;
       }
     }
+    if (SF->hasSemanticsAttr("string.equals")) {
```

Comment:
This has quadratic complexity.
Let's assume the function has many basic blocks and also many string comparisons. Then complexity is #blocks * #comparisons. 

Either you make this is separate function pass which scans the function a single time or you only handle this specific apply instruction in one run of the algorithm.

Reply:
I think I'll move it into a separate function pass and exit if `tyrOptimizeStringSwitchIn` returns `true`.